game-core/src/main/java/games/strategy/engine/data/RepairRuleList.java

```diff
@@ -4,6 +4,9 @@
 import java.util.HashMap;
 import java.util.Map;
 
+/**
+ * A keyed collection of {@link RepairRule}s.
```

Comment:
Knowing what the key value represents would help this further. 

Also of note, the comment says 'collection', but the class is suffixed with `List`, is it significant that this is an ordered collection?

Reply:
There is no order as the implementation uses a `HashMap` to store the repair rules.  Renaming the class isn't possible in order to maintain backwards compatibility.

Reply:
Okay, can we add a TODO to note that the class is badly named, the class is named "*List", the List data structure is ordered.

Still remains a mystery what the 'key' is, could you please document what we are using for keys?

Reply:
My thinking here, if we rename this to RepairRuleMap, then the current proposed javadoc would not be useful. But if we describe how the map is keyed, then such a javadoc would remain useful after the rename. The litmus test of "can we change the code so this comment is redundant", is a good test IMO for determining which comments should be deleted vs which ones are giving insights you would not get just by an initial reading of the code.

Reply:
> Still remains a mystery what the 'key' is, could you please document what we are using for keys?

Already done in 4323bed20.

> Okay, can we add a TODO to note that the class is badly named, the class is named "*List", the List data structure is ordered.

cb522ec84